restorer.go

```diff
@@ -1,36 +1,53 @@
 package lifecycle
 
 import (
+	"fmt"
+	"path/filepath"
+
 	"github.com/pkg/errors"
 	"golang.org/x/sync/errgroup"
 
+	"github.com/buildpacks/imgutil"
+
 	"github.com/buildpacks/lifecycle/buildpack"
+
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/launch"
 	"github.com/buildpacks/lifecycle/layers"
 	"github.com/buildpacks/lifecycle/platform"
 )
 
 type Restorer struct {
-	LayersDir  string
-	Buildpacks []buildpack.GroupBuildpack
-	Logger     Logger
+	LayersDir   string
+	Buildpacks  []buildpack.GroupBuildpack
+	Logger      Logger
+	SkipLayers  bool
+	PlatformAPI *api.Version
 }
 
 // Restore attempts to restore layer data for cache=true layers, removing the layer when unsuccessful.
 // If a usable cache is not provided, Restore will remove all cache=true layer metadata.
-func (r *Restorer) Restore(cache Cache) error {
+func (r *Restorer) Restore(img imgutil.Image, cache Cache) error {
 	// Create empty cache metadata in case a usable cache is not provided.
-	var meta platform.CacheMetadata
-	if cache != nil {
-		var err error
-		if !cache.Exists() {
-			r.Logger.Info("Layer cache not found")
+	meta, err := r.retrieveMetadataFrom(cache)
+	if err != nil {
+		return err
+	}
+
+	if r.PlatformAPI.Compare(api.MustParse("0.6")) >= 0 { // platform API >= 0.6
+		var appMeta platform.LayersMetadata
+		// continue even if the label cannot be decoded
+		if err := DecodeLabel(img, platform.LayerMetadataLabel, &appMeta); err != nil {
```

Comment:
Don't we already do this in analyzer?

Reply:
That is a good point. I guess we should read this from `analyzed.toml` here yeah?